feedhenry/fh-mbaas
# fix: Developement image failing to build on openshift:RHMAP-20656

# Jira link(s)
https://issues.jboss.org/browse/RHMAP-20656

# What
Update Dockerfile.dev to use forever npm, and change to node 6

# Why
npm install -g nodemon fails to install on Openshift 

```bash
/opt/rh/rh-nodejs6/root/usr/lib/node_modules/configstore/index.js:53
--
throw err;
^
Error: EACCES: permission denied, open '/opt/app-root/src/.config/configstore/update-notifier-nodemon.json'
You don't have access to this file.

```
# How
Change to the forever package


# Verification Steps
On a 4x cluster
1. Generate the config file: `grunt fh-generate-dockerised-config`
1. `docker build -t docker.io/my-Username/fh-mbaas:dev -f Dockerfile.dev .`
1. `docker push docker.io/my-Username/fh-mbaas:dev` 
1. change the image to your image from dockerhub
![change-image](https://user-images.githubusercontent.com/16667688/40659904-7d32fcd2-6347-11e8-9497-3d5fba023167.gif)
1. From the root of the `fh-mbaas directory, run the following:
```
oc login
oc project 3-node-mbaas
oc rsync --watch --no-perms=true ./lib $(oc get po | grep mbaas | grep Running | awk '{print $1}'):/opt/app-root/src
```
Check your pod has updated by going to the pod terminal and cd to /opt/app-root/src and cat a file you have changed to see if your changes have taken effect. Also you will see app restart in the logs.

```
error: restarting script because change changed
error: Forever detected script was killed by signal: SIGKILL
error: Script restart attempt #1
```

## Checklist:
